modules/polls/polls.js

```diff
@@ -0,0 +1,251 @@
+import { getLogger } from 'jitsi-meet-logger';
+import * as XMPPEvents from '../../service/xmpp/XMPPEvents';
+import * as JitsiConferenceEvents from '../../JitsiConferenceEvents';
+import { Strophe } from 'strophe.js';
+
+const logger = getLogger(__filename);
+
+const MESSAGE_POLL_END = 'poll-end';
+const MESSAGE_POLL_STARTED = 'poll-started';
+const MESSAGE_POLL_VOTE = 'poll-vote';
+
+
+/**
+ * A module for managing poll session related functionality.
+ * This module handles communication with all participants regarding the poll
+ * session management. It also keep the record of what poll is going
+ * on using the server side prosody-module if it is present.
+ */
```

Comment:
I really appreciate how you tried to encapsulate polls.

Reply:
speaking of such, currently assigning IDs for example happens actually in the UI. I think it should be moved down here, right? Is there a way to generate unique IDs in lib-jitsi-meet. I didn't want to add a new dependency package.

Reply:
What about prosody? Could that store the source of truth for all polls in the muc? It has uuid capabilities. Boris will also be reviewing your PRs and he's more knowledge about that side of the world.

Reply:
Yeah currently prosody module store that but it only stores it and tell people when they join about it.